### DISCUSSION TITLE
Status of 3.13 Release?

### DISCUSSION THREAD

**Discussion by richb-hanover** [sqlitebrowser/sqlitebrowser#3525] [General]
### Describe the new feature

Back in September 2023, it was announced that version 3.13 was ready for translation.

Do you have a projected release date? There's lots of good stuff that has evolved since 3.12... Thanks

**Comment by justinclift** [comment]
We've created some initial "Release Candidate 1" type of packages (a few weeks ago), but our windows signing certificate stuff has expired so we're getting that sorted out. :smile:

Should only be a few more weeks. :smile:

**Reply by Spatino19** [reply]
Is the RC1 still fine to use? It's been a few months so I don't know if it's stable enough.

**Reply by justinclift** [reply]
Ahhh.  Grab the build from here for now then.  It's just missing a translation update for one language from what will be the final thing (in theory):

https://github.com/sqlitebrowser/sqlitebrowser/releases/tag/v3.13.x-rc1

Note that if you're on windows though, it'll probably give you some warning about the package not being signed by a known developer.  That's what we're still waiting on the external crowd for. :wink:

**Reply by Spatino19** [reply]
Thank you

**Comment by richb-hanover** [comment]
@justinclift - any news on a 3.13 release version? What would prevent you from releasing in the next week? (Thanks, and I mean this entirely respectfully... If it's a single translation that's missing, you could include that item in the CHANGELOG...)

**Comment by justinclift** [comment]
We're waiting on part of the Windows binaries signing process.

**Comment by richb-hanover** [comment]
Thanks for that update. I have to say that the RC1 version has worked perfectly for my (limited) use case and testing.

There has been a lot of work since the RC1 release. Would you consider branching and creating a RC2 tag to "memorialize" the current state of the software? That would also let you rehearse the "release procedures" that need to happen when the Windows version does get signed.

Thank you again for this delightful piece of software.

**Reply by lucydodo** [reply]
I apologize for the delayed release.
Recently we've been working on some changes to build process that we need to make, and I'm hoping to have that done by the end of the month at the latest, at which point we'll be able to consider a RC2 release, and maybe talk about a stable release again.

**Reply by richb-hanover** [reply]
No apologies necessary, and I want to say this is terrific news. Thanks!

**Comment by richb-hanover** [comment]
One more thought: I just noticed the chronological list of updates (2023, 2022, 2021...) on the [Home page](https://sqlitebrowser.org/). 

I would encourage you to keep that list up-to-date, at least with brief entries for:

- 2023: 3.13-rc1 version available (This is a big deal!)
- 2024: Working on code signing for Windows binary using signpath.io

Those notes would assure (panicky / whiny) people (like me) that there is still progress. Thanks again.

**Reply by justinclift** [reply]
Heh Heh Heh.  Any interest in making a PR with those additions?

Website repo is here: https://github.com/sqlitebrowser/website

:grin:

**Reply by richb-hanover** [reply]
Yes - I left a couple questions at https://github.com/sqlitebrowser/website/issues/22

**Comment by lucydodo** [comment]
It's been a while, so I thought you might be interested to know.

Now all we need to do is include the 'sqlean' library in the Windows app bundle and do a final review of the changelog.
Barring any issues, we'll be releasing it sometime this month.

**Comment by richb-hanover** [comment]
This is DELIGHTFUL news. Thanks for the update. And congratulations - it has been a long slog.

**Reply by lucydodo** [reply]
Yes, I apologize for the delay, and we'll release next version as soon as possible. :)

**Comment by richb-hanover** [comment]
No apologies necessary.  Thanks again

**Comment by lucydodo** [comment]
Finally, a new version has been released: https://github.com/sqlitebrowser/sqlitebrowser/releases/tag/v3.13.0 :rocket:

**Comment by richb-hanover** [comment]
@lucydodo @justinclift @mgrojo - Congratulations all for pushing this over the finish line! I hope you feel great!

**Reply by lucydodo** [reply]
I really want to thank you personally, your contributions and interest have kept me motivated, especially the recent work on 'CHANELOG', which has helpme our to write better release notes. Thank you. Have a great day.

**Comment by richb-hanover** [comment]
You are very welcome. 

PS Do you plan to update the sqlitebrowser.org website to mention this new version? (I could submit a PR...)

**Reply by lucydodo** [reply]
Thank you, but that change is already in place and under review: https://github.com/sqlitebrowser/website/tree/v3.13.0

**Reply by justinclift** [reply]
Oops, I totally forgot this existed.  Looking at it now...

**Reply by justinclift** [reply]
k.  That's been merged.  Sorry for forgetting about it, the last few days have been a bit mentally fuzzy for me. :frowning:

**Reply by justinclift** [reply]
Ideally we'll add a blog entry about the new release to the sqlitebrowser.org website.

Generally the release announcement info on our website is a fully worked blog post that's effectively a copy of the release announcement here on GitHub.  i.e. same info, two different locations

@richb-hanover Any interest in making such a blog post on the website? :smile:

**Reply by richb-hanover** [reply]
@justinclift I think you're close to done. Here's what I would propose:

1. The "tag note" (at https://github.com/sqlitebrowser/sqlitebrowser/releases/tag/v3.13.0) **already is the blog post.** Move it there. 
2. Shorten that tag note to the essentials and point readers to the full blog post
3. Add `**New Version 3.13.0 - read the blog post**` just below the headline on the home page so that occasional visitors to the site will learn about the new version. (This could go away in 30 to 60 days)
4. Rename the "About" tab (in the header) to "History" and move it between "Stats" and "Patreon" tabs
5. When you're feeling bold, post an announcement to [Hacker News](https://news.ycombinator.com/submit) :-)

**Reply by justinclift** [reply]
> Rename the "About" tab (in the header) to "History" and move it between "Stats" and "Patreon" tabs

???

That seems weird.  The "About" page for a project traditionally tells people about the project.  Seems like a better name than History. :smile:

**Reply by chrisjlocke** [reply]
> @richb-hanover Any interest in making such a blog post on the website? 😄

I don't mind doing that if you want .. and slapping something on Twitter.

**Reply by justinclift** [reply]
@chrisjlocke Looks like it's all yours. :smile:

**Comment by richb-hanover** [comment]
> That seems weird. The "About" page for a project traditionally tells people about the project...

Hmmm... My initial impression of the page was that it cataloged the history of changes. I also entertained the thought of naming it "Credits" instead of "About". But I don't feel strongly about this.

If I were to make one change, I'd substitute "DB Browser for SQLite (DB4S)" for "This program" at the start of the page.

What about the other suggestions: treating the "tag note" as the new blog entry; adding "New Version" at the top of the home page; Hacker News? Thanks again.

**Reply by richb-hanover** [reply]
And I'm happy to let @chrisjlocke make suitable changes, and post to eX-Twitter.

**Comment by deepsidhu1313** [comment]
Is someone looking into updating this file https://github.com/sqlitebrowser/sqlitebrowser/blob/master/currentrelease, our snap packaging uses this for versioning. Can we add this in pre-release checklist for future ?

**Reply by lucydodo** [reply]
The file will be updated someitme next week. (Delayed Rollout)
We intentionally update late to avoid any potential critical issues in the new relesae.

**Reply by justinclift** [reply]
Yeah.  We tend to wait a few days from the initial release just in case there's some critical issue that we hadn't spotted.  It's happened before. :wink:

When that file is updated (both here in the repo and on our download server), it triggers the majority of our existing users to upgrade too.

**Reply by lucydodo** [reply]
Sorry for late, `currentrelease` file has just been updated.

**Reply by justinclift** [reply]
Heh Heh Heh.  That's only one of the two places it needs to be done. :wink:

The `currentrelease` file in the DB4S repo is only checked by really old versions of DB4S.  Modern DB4S instead checks https://download.sqlitebrowser.org/currentrelease, which still needs updating:

```
$ curl https://download.sqlitebrowser.org/currentrelease
3.12.2
https://sqlitebrowser.org/blog/version-3-12-2-released
```

To change that, update the string here then run the `deploy.sh` script in the root user home directory on the download server:

https://github.com/sqlitebrowser/db4s_cluster_downloader/blob/5626f6687b8ac8d0d3a6c555e1c34ce4af5fd398/main.go#L199

**Reply by justinclift** [reply]
Since it's a simple change, I just did it and updated the running server too.  It's now live. :smile:

**Comment by RRROMAN123** [comment]
It crashes constantly after changing the number of fields in the table.
And so it became faster to work.
You can't use this version yet
3.13 - beta version!

**Reply by justinclift** [reply]
How are you changing the number of fields in the table?

**Reply by richb-hanover** [reply]
@RRROMAN123 If this is still a concern, please open a new issue. Thanks
